write-fonts/src/font_builder.rs

```diff
@@ -15,6 +15,36 @@ pub struct FontBuilder<'a> {
     tables: BTreeMap<Tag, Cow<'a, [u8]>>,
 }
 
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub enum BuildFontError {
+    TooManyTables,
+}
+
+impl TableDirectory {
+    pub fn from_table_records(
+        table_records: Vec<TableRecord>,
+    ) -> Result<TableDirectory, BuildFontError> {
+        if table_records.len() > u16::MAX as usize {
```

Comment:
this seems so impossible to do in real use that I would be content with an assert; then we could forget about returning a `Result` for now, avoiding needing to do a minor version bump?
My medium-term vision is to get rid of this current `FontBuilder` entirely, and replace it with something where can just add any `TopLevelTable` object (which knows its tag) and then we would store these uncompiled, and compile them in parallel on build; this would also give us somewhere to do a font-wide validation pass (#321)

Reply:
I'm ok with dropping result here. I think the new API will need a Result then.

Reply:
Indeed that's very close to what FontBuilder *is* at this point; if build can have errors it can then do validation?

Reply:
No more result.